share/completions/fastboot.fish

```diff
@@ -0,0 +1,78 @@
+function __fish_fastboot_no_subcommand -d 'Test if fastboot has yet to be given the subcommand'
+    for i in (commandline -opc)
+        if contains -- $i flashall getvar oem flashing reboot update erase format devices flash get_staged help stage boot fetch
+            return 1
+        end
+    end
+    return 0
+end
+
+function __fish_fastboot_list_partition
+    set -l partitions boot bootloader dtbo modem odm odm_dlkm oem product pvmfw radio recovery system vbmeta vendor vendor_dlkm cache userdata system_ext
+    for i in $partitions
+        echo $i
+    end
+end
+
+function __fish_fastboot_list_partition_and_file
+    for i in (commandline -opc)
+        if contains -- (string replace -r '_[ab]' '' $i) (__fish_fastboot_list_partition)
```

Comment:
I'm curious, why the `_[ab]`?

Reply:
In newer devices with A/B (Seamless) System Updates, that partitions are divided into foo_a and foo_b, for example, boot_a, boot_b. Then when boot_a and boot_b and boot (represent current slot) is typed, it represent the target patition.

Reply:
![image](https://user-images.githubusercontent.com/12210746/165384648-9d75da2d-af2b-4ec1-8cbc-254645cdea9a.png)